flash-patcher.py

```diff
@@ -160,55 +193,153 @@ def inject(self):
                 split_line = line_stripped.split()
 
                 # Handle internal commands
-                if split_line[0] == "//" and split_line[1] == "cmd:":
-                    if split_line[2] == "skip":
-                        try:
-                            file_line_no += int(split_line[3])
-                            continue
-                        except ValueError:
-                            perror("")
-                            perror("Invalid skip amount: " + split_line[3])
-                            perror("Expected integer (got type \"str\")")
-                            perror("Aborting...")
-                            exit(1)
+                if split_line[:2] == ["//", "cmd:", "skip"]:
+                    n_str = split_line[3]
+                    try:
+                        n = int(n_str)
+                        continue
+                    except ValueError:
+                        exception(
+                            "Invalid skip amount: ",
+                            n,
+                            "\n",
+                            "Expected integer.\n",
+                            "Aborting...",
+                        )
+                        sys.exit(1)
+
+                    file_line_no += n
```

Comment:
move this inside the try/catch block instead of having it after the `except`

Reply:
It's like that in order to minimize the amount of processing done within the `try` block. If an unrelated `ValueError` were to somehow occur during the add-assign, we'd want to handle that separately.

(That said, the above logging statement is improperly formed - I should fix that.)